### PR TITLE
refactor(ava/data): migrate and improve dataAnalysis

### PR DESCRIPTION
**Pull request by pddpd**
Migrate and improve dataAnalysis, and add tests for it.

### CODE REVIEW

File: packages/ava/src/data/statistics/index.ts [hunk 1]
```diff
@@ -251,9 +251,9 @@ export function missing(value: unknown[]): number {
  * Calculate the counts of each distinct value in the array.
  * @param value - The array to process
  */
-export function valueMap(value: (string | number)[]): Record<string, number> {
+export function valueMap(value: unknown[]): Record<string, number> {
   const data: Record<string | number, number> = {};
-  value.forEach((v) => {
+  value.forEach((v: string | number) => {
```
Comment (Esorakouki) [inline packages/ava/src/data/statistics/index.ts:256]: is there any ts error without type assertions ?     unknown => string | number

File: packages/ava/tsconfig.json [hunk 2]
```diff
@@ -5,5 +5,5 @@
     "declaration": true,
     "declarationDir": "./lib"
   },
-  "include": ["./src"]
+  "include": ["./src", "__tests__/data"]
```
Comment (neoddish) [inline packages/ava/tsconfig.json:8]:
```suggestion
  "include": ["__tests__"]
```

File: packages/data-wizard/src/analyzer/index.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export interface FieldInfo {
   type: TypeSpecifics | 'mixed'; // float integer bool date null string mixed
   /** recommendation type */
   recommendation: TypeSpecifics;
-  /** number of empty inclues null undefined or empty string */
+  /** number of empty includes null undefined or empty string */
```
Comment (neoddish) [inline packages/data-wizard/src/analyzer/index.ts:25]: 6666666

File: packages/ava/src/data/statistics/index.ts (hunk 1 above)
Comment (pddpd) [reply packages/ava/src/data/statistics/index.ts]: fixed

File: packages/ava/tsconfig.json (hunk 2 above)
Comment (pddpd) [reply packages/ava/tsconfig.json]: fixed. 🤔 I have no impression. It may be an auto change.

**Review by neoddish — APPROVED** [review]
lgtm

**Pull request merged**